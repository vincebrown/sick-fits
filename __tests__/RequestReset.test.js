// Packages
import { mount } from 'enzyme'
import wait from 'waait'
import toJSON from 'enzyme-to-json'
import { MockedProvider } from 'react-apollo/test-utils'

// Components
import RequestReset, {
  REQUEST_RESET_MUTATION,
} from '../components/RequestReset'

const mocks = [
  {
    request: {
      query: REQUEST_RESET_MUTATION,
      variables: { email: 'vince@vincebrown.me' },
    },
    result: {
      data: { requestReset: { message: 'success', __typename: 'Message' } },
    },
  },
]

describe('<RequestReset/>', () => {
  it('renders and matches snapshot', async () => {
    const wrapper = mount(
      <MockedProvider>
        <RequestReset />
      </MockedProvider>
    )
    const form = wrapper.find('form[data-test="form"]')
    expect(toJSON(form)).toMatchSnapshot()
  })

  it('calls the mutation', async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <RequestReset />
      </MockedProvider>
    )
    // simulate typing an email.
    wrapper.find('input').simulate('change', {
      target: { name: 'email', value: 'vince@vincebrown.me' },
    })
    wrapper.find('form').simulate('submit')
    await wait()
    wrapper.update()
    const successMessage = wrapper.find('p')
    expect(toJSON(successMessage)).toMatchSnapshot()
  })
})
