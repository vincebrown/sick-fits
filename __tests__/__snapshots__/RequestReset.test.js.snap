// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset/> calls the mutation 1`] = `
<p>
  Success! Check your email for a reset link
</p>
`;

exports[`<RequestReset/> renders and matches snapshot 1`] = `
<form
  className="Form-sc-16eu58y-0 kdiGKW"
  data-test="form"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Request a Password Reset
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label
      htmlFor="email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="email"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Request Reset
    </button>
  </fieldset>
</form>
`;
